core/src/allocation_init.jl

```diff
@@ -579,15 +622,67 @@ function add_constraints_flow_conservation!(
             F[(inneighbor_id, node_id)] for
             inneighbor_id in inflow_ids_allocation(graph, node_id)
         ]),
-        base_name = "flow_conservation",
+        base_name = "flow_conservation_basin"
+    )
+
+    # Subnetwork inlets from main network
+    node_ids_inlets = [
+        edge[2] for edge in get_main_network_connections(p, allocation_network_id) if
+        edge[2] ∉ node_ids_basin
+    ]
+    problem[:flow_conservation_inlets] = JuMP.@constraint(
+        problem,
+        [node_id = node_ids_inlets],
+        sum([
+            F[(node_id, outneighbor_id)] for
+            outneighbor_id in outflow_ids_allocation(graph, node_id)
+        ]) == sum([
+            F[(inneighbor_id, node_id)] for
+            inneighbor_id in inflow_ids_allocation(graph, node_id)
+        ]),
+        base_name = "flow_conservation_inlet"
+    )
+
+    # Nodes with flow demand
+    F_flow_buffer_in = problem[:F_flow_buffer_in]
+    F_flow_buffer_out = problem[:F_flow_buffer_out]
+    node_ids_flow_demand = [
+        node_id for
+        node_id in node_ids if has_external_demand(graph, node_id, :flow_demand)[1]
+    ]
+    problem[:flow_conservation_flow_demand] = JuMP.@constraint(
+        problem,
+        [node_id = node_ids_flow_demand],
+        F[(node_id, only(outflow_ids_allocation(graph, node_id)))] +
+        F_flow_buffer_in[node_id] ==
+        F[(only(inflow_ids_allocation(graph, node_id)), node_id)] +
+        F_flow_buffer_out[node_id],
+        base_name = "flow_conservation_flow_demand"
+    )
+
+    # Subnetwork inlets from main network
```

Comment:
And also for this one.

Reply:
Looks like this was a duplicate of the first codeblock, can be removed altogether